hack/util.sh

```diff
@@ -8,116 +8,174 @@ TIME_MIN=$((60 * $TIME_SEC))
 # setup_env_vars exports all the necessary environment variables for configuring and
 # starting OS server.
 function setup_env_vars {
-  export ETCD_DATA_DIR="${BASETMPDIR}/etcd"
-  export VOLUME_DIR="${BASETMPDIR}/volumes"
-  export FAKE_HOME_DIR="${BASETMPDIR}/openshift.local.home"
-  export API_HOST="${API_HOST:-127.0.0.1}"
-  export API_PORT="${API_PORT:-8443}"
-  export API_SCHEME="${API_SCHEME:-https}"
-  export MASTER_ADDR="${API_SCHEME}://${API_HOST}:${API_PORT}"
-  export PUBLIC_MASTER_HOST="${PUBLIC_MASTER_HOST:-${API_HOST}}"
-  export KUBELET_SCHEME="${KUBELET_SCHEME:-https}"
-  export KUBELET_HOST="${KUBELET_HOST:-127.0.0.1}"
-  export KUBELET_PORT="${KUBELET_PORT:-10250}"
-  export SERVER_CONFIG_DIR="${BASETMPDIR}/openshift.local.config"
-  export MASTER_CONFIG_DIR="${SERVER_CONFIG_DIR}/master"
-  export NODE_CONFIG_DIR="${SERVER_CONFIG_DIR}/node-${KUBELET_HOST}"
-
-  # set path so OpenShift is available
-  GO_OUT="${OS_ROOT}/_output/local/go/bin"
-  export PATH="${GO_OUT}:${PATH}"
+	# determine API_HOST.  This is used to derive other values.  Bye default, it is first ipv4 address that isn't loopback or docker bridge
+	DEFAULT_SERVER_IP=`ifconfig | grep -Ev "(127.0.0.1|172.17.42.1)" | grep "inet " | head -n 1 | sed 's/adr://' | awk '{print $2}'`
+	export API_HOST="${API_HOST:-${DEFAULT_SERVER_IP}}"
+
+	export LOG_DIR="${LOG_DIR:-${BASETMPDIR}/logs}"
+	export ETCD_DATA_DIR="${BASETMPDIR}/etcd"
+	export VOLUME_DIR="${BASETMPDIR}/volumes"
+	export FAKE_HOME_DIR="${BASETMPDIR}/openshift.local.home"
+	export API_PORT="${API_PORT:-8443}"
+	export API_SCHEME="${API_SCHEME:-https}"
+	export MASTER_ADDR="${API_SCHEME}://${API_HOST}:${API_PORT}"
+	export PUBLIC_MASTER_HOST="${PUBLIC_MASTER_HOST:-${API_HOST}}"
+	export KUBELET_SCHEME="${KUBELET_SCHEME:-https}"
+	export KUBELET_HOST="${KUBELET_HOST:-127.0.0.1}"
+	export KUBELET_PORT="${KUBELET_PORT:-10250}"
+	export SERVER_CONFIG_DIR="${BASETMPDIR}/openshift.local.config"
+	export MASTER_CONFIG_DIR="${SERVER_CONFIG_DIR}/master"
+	export NODE_CONFIG_DIR="${SERVER_CONFIG_DIR}/node-${KUBELET_HOST}"
+	export ARTIFACT_DIR="${ARTIFACT_DIR:-${BASETMPDIR}/artifacts}"
+
+	# Use either the latest release built images, or latest.
+	if [[ -z "${USE_IMAGES-}" ]]; then
+		IMAGES='openshift/origin-${component}:latest'
+		export USE_IMAGES=${IMAGES}
+		if [[ -e "${OS_ROOT}/_output/local/releases/.commit" ]]; then
+			COMMIT="$(cat "${OS_ROOT}/_output/local/releases/.commit")"
+			IMAGES="openshift/origin-\${component}:${COMMIT}"
+			export USE_IMAGES=${IMAGES}
+		fi
+	fi
+
+	if [[ "${API_SCHEME}" == "https" ]]; then
+		export CURL_CA_BUNDLE="${MASTER_CONFIG_DIR}/ca.crt"
+		export CURL_CERT="${MASTER_CONFIG_DIR}/admin.crt"
+		export CURL_KEY="${MASTER_CONFIG_DIR}/admin.key"
+	fi
+
+	# change the location of $HOME so no one does anything naughty
+	export HOME="${FAKE_HOME_DIR}"
```

Comment:
> Why is the mkdir not necessary here?

Trying to keep the side effects of this function narrow.  Once you have the value, you can decide later on what action you'd like to take.  For instance, maybe someone is calling this to build the expected env for data gathering of some kind.